soci/soci_index.go

```diff
@@ -160,7 +160,7 @@ func BuildSociIndex(ctx context.Context, cs content.Store, img images.Image, spa
 		eg.Go(func() error {
 			desc, err := buildSociLayer(ctx, cs, l, spanSize, store, &config)
 			if err != nil {
-				return err
+				return fmt.Errorf("could not build zTOC for %s: %w", l.Digest.String(), err)
```

Comment:
could this cause a runtime error based on whether `l` has `l.Digest.String()`?

Reply:
No, `l.Digest` is not a pointer so it must be present. 

It also wouldn't be a valid descriptor, so it's not clear that it would even be possible to get containerd to accept a manifest like that. And this also isn't the first access to `l.Digest`. `buildSociLayer` calls `contentStore.ReaderAt` which uses the digest to find open the file on disk.